# GCC 8.3 (tg5040) fails class template argument deduction for
# `new IndependentTask(type, prio, lambda)` in Core/Screenshot.cpp; a factory
# function template deduces fine on every compiler.
diff --git a/Common/Thread/Promise.h b/Common/Thread/Promise.h
index 9a98ac1..21d7a1c 100644
--- a/Common/Thread/Promise.h
+++ b/Common/Thread/Promise.h
@@ -23,6 +23,12 @@ class IndependentTask : public Task {
 	TaskPriority prio_;
 };
 
+// GCC 8 cannot deduce IndependentTask<T> in a new-expression; a factory function can.
+template <class T>
+IndependentTask<T> *MakeIndependentTask(TaskType type, TaskPriority prio, T func) {
+	return new IndependentTask<T>(type, prio, std::move(func));
+}
+
 template<class T>
 class PromiseTask : public Task {
 public:
diff --git a/Core/Screenshot.cpp b/Core/Screenshot.cpp
index 38662b2..7a00874 100644
--- a/Core/Screenshot.cpp
+++ b/Core/Screenshot.cpp
@@ -393,7 +393,7 @@ bool ScreenshotNotifyPostGameRender(Draw::DrawContext *draw) {
 }
 
 static void SaveScreenshotAsync(GPUDebugBuffer &&buf, int w, int h, int maxRes) {
-	g_threadManager.EnqueueTask(new IndependentTask(TaskType::IO_BLOCKING, TaskPriority::LOW,
+	g_threadManager.EnqueueTask(MakeIndependentTask(TaskType::IO_BLOCKING, TaskPriority::LOW,
 		[filename = std::move(g_pendingScreenshot.path), buf = std::move(buf), callback = std::move(g_pendingScreenshot.callback), fmt = g_pendingScreenshot.format, w, h, maxRes]() {
 		u8 *flipbuffer = nullptr;
 		u32 width = w, height = h;
